### PR TITLE
qna authoring samples

### PR DESCRIPTION
**Pull request by mshaban-msft**

### CODE REVIEW

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        delete_poller = client.question_answering_projects.delete_project(
+            project_name=project_name
+        )
+        delete_poller.result()
+
+        # import project
+        import_poller = client.question_answering_projects.begin_import_method(
+            project_name=project_name,
+            body={
+                "file_uri":"asdsadasdasd"
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py:66]: This file uri should be retrieved from the previous export result.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_create_and_deploy_project.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        )
+        update_sources_poller.result()
+
+        # list sources
+        print("list project sources")
+        sources = client.question_answering_projects.get_sources(
+            project_name=project_name
+        )
+        for source in sources:
+            print(source) # needs to properly access values!
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_create_and_deploy_project.py:84]: So this list operation isn't functioning correctly?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                "multilingualResource": True,
+                "settings": {
+                    "defaultAnswer": "no answer"
+                }
+            })
+
+        # export
+        export_poller = client.question_answering_projects.begin_export(
+            project_name=project_name
+            # can't find models for other properties!!! (not even swagger definition is clear!)
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py:52]: So there's supposed to be a request body here?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        )
+        for item in qnas:
+            print(item)
+
+        # sources
+        sources_poller = client.question_answering_projects.begin_update_sources(
+            project_name=project_name,
+            body={
+                
+            }
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py:100]: No content?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py (hunk 1 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py]: yeah!
but there seems to be a problem with LLC generation ..
export function returns LROPoller[None]
this is why we can't retrieve result url

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_create_and_deploy_project.py (hunk 2 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_create_and_deploy_project.py]: updated

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py (hunk 3 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py]: updated

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py (hunk 4 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py]: updated

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/README.md [hunk 5]
```diff
@@ -129,6 +146,95 @@ output = await client.get_answers(
...
+            "description": "biography of Sir Issac Newton",
+            "language": "en",
+            "multilingualResource": True,
+            "settings": {
+                "defaultAnswer": "no answer"
+            }
+        })
+
+    print("view created project info:")
+    print(u"\tname: {}".format(project["projectName"]))
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/README.md:180]: We shouldn't need the `u` prefix - this is a Python 2.7 format that we no longer support.

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                }
+            })
+
+        # export
+        export_poller = client.begin_export(
+            project_name=project_name,
+            format="json"
+        )
+        export_result = export_poller.result()
+        export_url = export_result["resultUrl"]
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py:57]: What's the data at this URL? Could we maybe download it, then feed it into the subsequent import?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            print("\tquestions:")
+            for question in item["questions"]:
+                print(u"\t\t{}".format(question))
+            print(u"\tanswer: {}".format(item["answer"]))
+
+        # synonyms
+        client.update_synonyms(
+            project_name=project_name,
+            synonyms={
+                "value": [
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py:103]: Yeah it's kinda weird that this payload is so different to the other update operations

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py (hunk 6 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_export_import_project.py:57]: exactly!
this was our first intuition, to use this export url in the import method 
but the thing is, this url is key-protected, and we can only submit public urls for the import method
we thought of including a sample method to download the file form this url
we actually need to ask Rohan from the service team about this

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py (hunk 7 above)
Comment (mshaban-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py:103]: addressed this [here](https://github.com/Azure/azure-sdk-for-python/pull/21803#discussion_r795211708)

**Review by annatisch — APPROVED** [review]

**Pull request merged**